socket/src/commonTest/kotlin/mqtt/socket/SocketTests.kt

```diff
@@ -185,82 +142,117 @@ class SocketTests {
         }
     }
 
-    private suspend fun stressTest(
-        scope: CoroutineScope,
-        serverL: ServerLaunched?,
-        getServerSocket: () -> ServerSocket, getClientSocket: () -> ClientToServerSocket
+    @ExperimentalUnsignedTypes
+    private suspend fun stessDisconnectTest(
+        getServerSocket: () -> TCPServer, getClientSocket: () -> ClientToServerSocket
     ) {
-        val serverLaunched = serverL ?: launchServer(scope, getServerSocket)
-        repeat(clientCount.toInt()) {
-            try {
+        val clientDoneMutex = Mutex(locked = false)
+        var port: UShort = 0u
+        var counter = 0
+
+        try {
+            val server = launchServer(getServerSocket())
+
+            port = server.getListenPort()
+
+            clientDoneMutex.lock()
+            repeat (clientCount1.toInt()) {
                 val client = getClientSocket()
-                serverLaunched.serverClientSocket = client
-                client.open(connectTimeout, serverLaunched.serverSocket.port()!!)
-                serverLaunched.firstMessageReceivedLock.lock()
-                assertTrue(client.isOpen())
-                val clientPort = client.localPort()!!
-                assertNotNull(serverLaunched.server.connections[clientPort])
+                initiateClient(client, port)
+                assertTrue(client.isOpen(), "Client connection is not open")
                 client.close()
-                serverLaunched.server.closeClient(clientPort)
-                assertNull(serverLaunched.server.connections[clientPort])
-            } catch (e: Throwable) {
-                println("Failed to validate client #$it")
-                throw e
-            } finally {
-                if (validateCloseWaitAgressive) {
-                    checkPort(serverLaunched.server)
-                }
+                assertFalse(client.isOpen(), "Client connetion is still open")
+                counter++
+
+                if (counter >= clientCount1)
+                    clientDoneMutex.unlock()
             }
+
+            clientDoneMutex.lock()
+            assertTrue(server.isOpen(), "Server has closed")
+            server.close()
+            assertFalse(server.isOpen(), "Server is still open")
+
+        } catch (e: Exception) {
+            println("stessDisconnectTest.exception: ${e.message}")
```

Comment:
should we be catching this exception or passing it up?

Reply:
passing it up.